apps/web/pages/settings/my-account/out-of-office/index.tsx

```diff
@@ -86,26 +115,68 @@ const CreateOutOfOfficeEntryModal = ({
     },
   });
 
+  const resetForm = useCallback(() => {
+    reset();
+    setSelectedReason(null);
+    setSelectedMember(null);
+    setProfileRedirect(false);
+  }, [reset]);
+
   const createOutOfOfficeEntry = trpc.viewer.outOfOfficeCreate.useMutation({
     onSuccess: () => {
-      showToast(t("success_entry_created"), "success");
+      showToast(
+        currentlyEditingOutOfOfficeEntry
+          ? t("success_edited_entry_out_of_office")
+          : t("success_entry_created"),
+        "success"
+      );
       utils.viewer.outOfOfficeEntriesList.invalidate();
-      setProfileRedirect(false);
+      resetForm();
       closeModal();
     },
     onError: (error) => {
       showToast(t(error.message), "error");
     },
   });
 
-  const { data: outOfOfficeReasonList } = trpc.viewer.outOfOfficeReasonList.useQuery();
+  useEffect(() => {
+    if (currentlyEditingOutOfOfficeEntry === null) {
+      resetForm();
+      return;
+    }
 
-  const reasonList = [
-    ...(outOfOfficeReasonList || []).map((reason) => ({
-      label: `${reason.emoji} ${reason.userId === null ? t(reason.reason) : reason.reason}`,
-      value: reason.id,
-    })),
-  ];
+    const selectedReason =
+      reasonList.find((reason) => {
+        return reason.value === currentlyEditingOutOfOfficeEntry.selectedReason;
+      }) ?? null;
+    if (selectedReason && selectedReason.value) {
+      setValue("reasonId", selectedReason.value);
+      setSelectedReason(selectedReason);
+    }
+
+    const selectedMember =
+      memberListOptions.find((member) => {
+        return member.value === currentlyEditingOutOfOfficeEntry.selectedMember;
+      }) ?? null;
+
+    setProfileRedirect(currentlyEditingOutOfOfficeEntry.profileRedirect);
+
+    if (selectedMember && selectedMember?.value) {
+      setValue("toTeamUserId", selectedMember.value);
+      setSelectedMember(selectedMember);
+    }
+
+    if (currentlyEditingOutOfOfficeEntry.notes) {
+      setValue("notes", currentlyEditingOutOfOfficeEntry.notes);
+    }
+
+    setValue("dateRange", {
+      startDate: dayjs(currentlyEditingOutOfOfficeEntry.start).startOf("d").toDate(),
+      endDate: dayjs(currentlyEditingOutOfOfficeEntry.end).subtract(1, "d").endOf("d").toDate(),
+    });
+
+    setValue("uuid", currentlyEditingOutOfOfficeEntry.uuid);
+  }, [currentlyEditingOutOfOfficeEntry, setValue, reasonList, dateRange, memberListOptions, resetForm]);
```

Comment:
It looks not clean to me. There are improvements needed. Let's keep it straightforward. What if you made an EditOOO component, then you pass the item or required props when you edit an OOO list item? Set the form `defaultValues` to it. Can you not do the same here? Instead of using too many `setValue` calls, I know it's not straightforward; some field values need to be extracted from the ID, but it's not difficult. The code would be shorter and clearer. Think of it as a universal component and consider what you would do if there were individual components. Let me know what you think.